src/Client.php

```diff
@@ -301,6 +323,28 @@ protected function getContext(ConnectionInterface $connection)
     }
 
     /**
+     * Derives a set of socket context options for a given secure connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return array Associative array of context option key-value pairs
+     */
+    protected function getSecureContext(ConnectionInterface $connection)
+    {
+        $context = array();
+        if ($this->getForceIpv4Flag($connection)) {
+            $context['bindto'] = '0.0.0.0:0';
+        }
+        if ($this->getVerifyPeerFlag($connection)) {
+            $context['verify_peer'] = false;
+        }
+        if ($this->getAllowSelfSignedFlag($connection)) {
+            $context['allow_self_signed'] = true;
+        }
+        //$context = array('socket' => $context);
```

Comment:
You're correct, `SecureConnector` appears to [expect socket options specifically](https://github.com/reactphp/socket-client/blob/v0.5.3/src/SecureConnector.php#L48). However, it also appears that `TcpConnector` [does as well](https://github.com/reactphp/socket-client/blob/v0.5.3/src/TcpConnector.php#L36). As such, this PR will need to update [this line](https://github.com/mvangoor/phergie-irc-client-react/blob/71bac148f758f3c930fa0a8c32f006dbd40b9a55/src/Client.php#L321) as well to avoid breaking non-secure connections.